src/Workspaces/CoreTest/WorkspaceTests/AdhocWorkspaceTests.cs

```diff
@@ -2,26 +2,43 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System;
+using System.Collections.Immutable;
 using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Shared.Utilities;
 using Microsoft.CodeAnalysis.Test.Utilities;
 using Microsoft.CodeAnalysis.Text;
+using Microsoft.CodeAnalysis.UnitTests.Persistence;
 using Roslyn.Test.Utilities;
 using Roslyn.Utilities;
-using System.Threading;
-using System.Threading.Tasks;
 using Xunit;
-using Microsoft.CodeAnalysis.Shared.Utilities;
-using System;
 using CS = Microsoft.CodeAnalysis.CSharp;
-using Microsoft.CodeAnalysis.Host.Mef;
-using System.Collections.Immutable;
 
 namespace Microsoft.CodeAnalysis.UnitTests
 {
     [UseExportProvider]
     public partial class AdhocWorkspaceTests : TestBase
     {
+        private static AdhocWorkspace CreateWorkspace(Type[] additionalParts = null)
+            => new AdhocWorkspace(FeaturesTestCompositions.Features.WithAdditionalParts(additionalParts).GetHostServices());
+
+        private static AdhocWorkspace CreateWorkspaceWithRecoverableSyntaxTrees()
```

Comment:
💡 If this is used more than once, it would be faster to cache the test composition with added parts in a temporary type.